src/aolayer.cpp

```diff
@@ -118,6 +118,18 @@ void AOLayer::move(int ax, int ay)
   QLabel::move(x, y);
 }
 
+void AOLayer::move_and_center(int ax, int ay)
+{
+  x = ax;
+  y = ay;
+  if (movie_frames.isEmpty()) // safeguard
+    QLabel::move(x,y);
+  else if (frame >= movie_frames.count())
+    set_frame(movie_frames[0]);
+  else
+    set_frame(movie_frames[frame]);
```

Comment:
Really smelly code. Probably should move the `QLabel::move` and `setMask` calls in `set_frame` to another method.

Reply:
i can't. both of those calls depend on being given a pixmap to work from and there isn't a way to properly reference the current pixmap of a QLabel

Reply:
actually, I don't need to reference the *current* pixmap if all I'm doing is centering it. since all pixmaps should be the same size, i can just reference frame 1 for the center operation